pyfar/dsp/dsp.py

```diff
@@ -811,6 +813,338 @@ def regularized_spectrum_inversion(
     return inverse
 
 
+class interpolate_spectrum():
+    """
+    Interpolate incomplete spectrum to complete single sided spectrum.
+
+    TODO: Add reference to minimum phase functions
+
+    Parameters
+    ----------
+    data : FrequencyData
+        Input data to be interpolated. `data.fft_norm` must be `'none'`.
+    method : string
+        Specifies the input data for the interpolation
+
+        ``'complex'``
+            Separate interpolation of the real and imaginary part
+        ``'magnitude_unwrap'``
+            Separate interpolation if the magnitude and unwrapped phase values
+        ``'magnitude_linear'``
+            Interpolation of the magnitude values and generation of a linear
+            phase response with a specified group delay. (See parameters
+            `group_delay` and `unit` below and
+            :py:func:`~pyfar.dsp.linear_phase`).
+        ``'magnitude_minimum'``
+            Interpolation of the magnitude values and generation of a minimum
+            phase response
```

Comment:
I have thought about this a bit:
These options are pretty much independent from the actual interpolation step, but apply further processing on the spectrum, that is they do phase manipulation, independent of the interpolation of the spectrum. For this reason I would suggest removing them from the interpolation. This would make this class much slimmer, since it already has quite a number of options already. We can add examples on how to generate a lin phase variant of the interpolated signal for example.

Reply:
Agreed - this was the old me trying to put too much into one function...